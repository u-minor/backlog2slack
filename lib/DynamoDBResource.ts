import { Stack } from 'aws-cdk-lib';
import { AttributeType, BillingMode, Table } from 'aws-cdk-lib/aws-dynamodb';

export class DynamoDBResource {
  table: Table;

  constructor(stack: Stack) {
    this.table = new Table(stack, 'CacheTable', {
      billingMode: BillingMode.PAY_PER_REQUEST,
      partitionKey: { name: 'key', type: AttributeType.STRING },
      timeToLiveAttribute: 'ttl',
    });
  }
}
