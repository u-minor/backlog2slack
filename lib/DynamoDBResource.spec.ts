import { App, Stack } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { DynamoDBResource } from './DynamoDBResource';

describe('DynamoDBResource', () => {
  let template: Template;

  beforeAll(() => {
    const app = new App();
    const stack = new Stack(app, 'TestStack');

    new DynamoDBResource(stack);

    template = Template.fromStack(stack);
  });

  it('should generate valid DynamoDB table resource', () => {
    const actual = template.findResources('AWS::DynamoDB::Table');

    expect(actual).toMatchSnapshot();
  });
});
