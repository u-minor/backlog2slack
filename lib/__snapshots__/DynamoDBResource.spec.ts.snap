// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DynamoDBResource should generate valid DynamoDB table resource 1`] = `
{
  "CacheTableC1E6DF7E": {
    "DeletionPolicy": "Retain",
    "Properties": {
      "AttributeDefinitions": [
        {
          "AttributeName": "key",
          "AttributeType": "S",
        },
      ],
      "BillingMode": "PAY_PER_REQUEST",
      "KeySchema": [
        {
          "AttributeName": "key",
          "KeyType": "HASH",
        },
      ],
      "TimeToLiveSpecification": {
        "AttributeName": "ttl",
        "Enabled": true,
      },
    },
    "Type": "AWS::DynamoDB::Table",
    "UpdateReplacePolicy": "Retain",
  },
}
`;
